### PR TITLE
Fix PD failover when multi-cluster enabled

### PR DESCRIPTION
**Pull request by handlerww**
### What problem does this PR solve? 
When multi-cluster enabled, PD member name is in long format(FQDN), some member selections in pd_failover are using pod name.

### What is changed and how does it work?
Fix codes to be compatible with the long format.
### Check List 

Tests 

- Unit test

Code changes

 - Has Go code change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: pkg/manager/member/pd_member_manager.go [hunk 1]
```diff
@@ -267,9 +267,10 @@ func (m *pdMemberManager) shouldRecover(tc *v1alpha1.TidbCluster) bool {
 		if !podutil.IsPodReady(pod) {
 			return false
 		}
-		status, ok := tc.Status.PD.Members[pod.Name]
-		if !ok || !status.Health {
-			return false
+		for pdName, pdMember := range tc.Status.PD.Members {
```
Comment (DanielZhangQD) [inline pkg/manager/member/pd_member_manager.go:270]: What if there is no member of this Pod?

Comment (handlerww) [reply pkg/manager/member/pd_member_manager.go:270]: If there is no member of this pod, there is no need to set off the failover function. When a member is reported as unhealthy, the controller begins to do failover operations.

Comment (handlerww) [reply pkg/manager/member/pd_member_manager.go:270]: It can be the launching stage.

File: pkg/manager/member/pd_member_manager.go [hunk 2]
```diff
@@ -267,8 +267,16 @@ func (m *pdMemberManager) shouldRecover(tc *v1alpha1.TidbCluster) bool {
...
 		}
-		status, ok := tc.Status.PD.Members[pod.Name]
-		if !ok || !status.Health {
+		ok := false
+		for pdName, pdMember := range tc.Status.PD.Members {
+			if strings.Split(pdName, ".")[0] == pod.Name {
+				if !pdMember.Health {
+					return false
+				}
+				ok = true
```
Comment (DanielZhangQD) [inline pkg/manager/member/pd_member_manager.go:276]: Should `break` here?

File: pkg/manager/member/pd_member_manager.go [hunk 3]
```diff
@@ -267,8 +267,17 @@ func (m *pdMemberManager) shouldRecover(tc *v1alpha1.TidbCluster) bool {
 		if !podutil.IsPodReady(pod) {
 			return false
 		}
-		status, ok := tc.Status.PD.Members[pod.Name]
-		if !ok || !status.Health {
+		ok := false
+		for pdName, pdMember := range tc.Status.PD.Members {
```
Comment (DanielZhangQD) [inline pkg/manager/member/pd_member_manager.go:271]: Please update UT to cover your code change.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by lonng — APPROVED** [review]
LGTM

**Pull request merged**